clients/roscpp/src/libros/service_server_link.cpp

```diff
@@ -362,15 +368,38 @@ bool ServiceServerLink::call(const SerializedMessage& req, SerializedMessage& re
     processNextCall();
   }
 
+  bool timed_out = false;
   {
     boost::mutex::scoped_lock lock(info->finished_mutex_);
 
+    using namespace boost::chrono;
+    steady_clock::time_point end_time = steady_clock::now() + nanoseconds(timeout.toNSec());
```

Comment:
This variable is only needed when `timeout > 0`. Please avoid calling `steady_clock::now()` if the result isn't used.